### ISSUE TITLE
Snapcraft documentation outdated

### ISSUE THREAD

**Issue by saylerb** [circleci/circleci-docs#6243]
## TL; DR

Following the [CircleCI docs for building a snap](https://circleci.com/docs/2.0/build-publish-snap-packages/) with a docker agent currently does not seem possible with `snapcraft` version 6. After hacking around trying to get it to work, we decided to use a machine agent for now.

## Description
Currently there doesn’t seem to be a maintained and up-to-date docker image for `snapcraft`:

- `cibuilds/snapcraft:stable` is deprecated and uses Ubuntu 16.04.6 LTS `snapcraft` v3.8, a version from September 2019. 

- We searched around the internet looking for another image that could be a suitable replacement. `snapcore/snapcraft` docker image uses Ubuntu 16.04 and has snapcraft version 4.6.x, which is also out of date but slightly newer than the one provided by cibuilds.

Our team has tried to create our own docker image with the Ubuntu 20 and snapcraft 6.0, which was difficult due to `snapd` not being able to run easily and requirement for some kind of python shim. Due to the absence of an actively maintained or updated `snapcraft` image, we decided to go with a machine agent. The docs probably should be updated to recommend the use of a machine agent as well, until a suitable image is found or the existing ones updated. 

## Location
[Docs for building and publishing a snap package](https://circleci.com/docs/2.0/build-publish-snap-packages/)

## Steps to Reproduce
```
docker run --rm -it cibuilds/snapcraft:stable snapcraft --version
docker run --rm -it snapcore/snapcraft:latest snapcraft --version
```

**Comment by devinmarieb** [comment]
@saylerb Thanks! We will update the docs and reach out to the appropriate engineering team to see if they have further recommendations. As a side note, I think I went to coding school in Denver with your brother 😄

**Comment by saylerb** [comment]
Small world @devinmarieb! Always a fun surprise when I get to collaborate with grads in the industry 😃  Our team may get some time with Canonical in the coming weeks so I might have some better ideas soon for what agent should be used. I can update here if I find out anything useful.
